erpnext_germany/erpnext_germany/doctype/business_trip/business_trip.js

```diff
@@ -84,3 +100,140 @@ frappe.ui.form.on("Business Trip Allowance", {
 		}
 	},
 });
+
+function show_processing_details_dialog(frm) {
+	frappe.call({
+		method: "erpnext_germany.erpnext_germany.doctype.business_trip.business_trip.get_processing_details",
+		args: {
+			business_trip: frm.doc.name,
+		},
+		callback: function (r) {
+			if (r.message) {
+				// Prepare table data first
+				let table_data = prepare_table_data(r.message);
+
+				// Create dialog with the data
+				let dialog = new frappe.ui.Dialog({
+					title: __("Processing Details"),
+					fields: [
+						{
+							fieldname: "processing_details",
+							fieldtype: "Table",
+							label: __("Linked Documents"),
+							cannot_add_rows: true,
+							in_place_edit: false,
+							reqd: 0,
+							data: table_data,
+							get_data: function () {
+								return this.data;
+							},
+							fields: [
+								{
+									fieldtype: "Link",
+									fieldname: "doctype",
+									label: __("DocType"),
+									options: "DocType",
+									read_only: 1,
+									in_list_view: 1,
+								},
+								{
+									fieldtype: "Dynamic Link",
+									fieldname: "document_name",
+									label: __("Document Name"),
+									options: "doctype",
+									read_only: 1,
+									in_list_view: 1,
+								},
+								{
+									fieldtype: "Data",
+									fieldname: "supplier_name",
+									label: __("Supplier Name"),
+									read_only: 1,
+									in_list_view: 1,
+								},
+								{
+									fieldtype: "Currency",
+									fieldname: "grand_total",
+									label: __("Grand Total"),
+									read_only: 1,
+									in_list_view: 1,
+								},
+								{
+									fieldtype: "Data",
+									fieldname: "status",
+									label: __("Status"),
+									read_only: 1,
+									in_list_view: 1,
+								},
+							],
+						},
+					],
+					size: "large",
+					primary_action_label: __("Close"),
+					primary_action: function () {
+						dialog.hide();
+					},
+				});
+
+				dialog.show();
+			} else {
+				frappe.msgprint(__("No processing details found."));
+			}
+		},
+	});
+}
+
+function prepare_table_data(data) {
+	let table_data = [];
+
+	if (data && data.length > 0) {
+		data.forEach(function (record) {
+			table_data.push({
+				doctype: __(record.doctype),
+				document_name: record.name,
+				grand_total: record.grand_total || 0,
+				status: __(record.status),
+				supplier_name: record.supplier_name || "",
+			});
+		});
+	}
+
+	if (table_data.length === 0) {
+		table_data.push({
+			doctype: "",
+			document_name: __("No linked documents found"),
+			grand_total: 0,
+			status: "",
+			supplier_name: "",
+		});
+	}
+
+	return table_data;
+}
+
+function create_purchase_invoice_with_receipt(frm, cdt, cdn) {
+	if (frm.is_dirty()) {
+		frappe.msgprint({
+			title: __("Save Required"),
+			message: __(
+				"Please save the Business Trip document first before creating a Purchase Invoice."
```

Comment:
```suggestion
				"Before creating a purchase invoice, please save this record."
```

Hauptsächlich "the Business Trip document" -> "this record", ist etwas kompakter. Aber sicher Geschmackssache.